includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -3082,9 +3082,36 @@ static function ( $selector_language ) {
 			// Replace focus selectors with :focus-within.
 			if ( $this->focus_class_name_selector_pattern ) {
 				$count    = 0;
-				$selector = preg_replace(
+				$selector = preg_replace_callback(
 					$this->focus_class_name_selector_pattern,
-					':focus-within',
+					static function ( $matches ) {
+						$replacement = ':focus-within';
+
+						if (
+							'focus' === $matches['class']
```

Comment:
I added this condition because this issue was only observed for core themes. Since `focus_within_classes` is a sanitizer arg, it could be that someone is filtering the class names to be something else and so it wouldn't apply, since it wouldn't be a core theme context. Granted, this is not a likely occurrence.